cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -438,6 +438,103 @@ func isNodeSelectorUpdated(old *v1.NodeSelector, new *v1.NodeSelector) bool {
 	return !reflect.DeepEqual(old.NodeSelectorTerms, new.NodeSelectorTerms)
 }
 
+// isProtocolConfigUpdated checks if protocol is updated
+func isProtocolConfigUpdated(old *v1alpha1.ProtocolConfig, new *v1alpha1.ProtocolConfig) bool {
+	return !reflect.DeepEqual(old, new)
+}
+
+// updateProtocolInConfigMap updates the protocol in the deviceProfile in configmap
+func (dc *DownstreamController) updateProtocolInConfigMap(device *v1alpha1.Device) {
```

Comment:
I suggest to add a return error for `updateProtocolInConfigMap()`.

Reply:
The functions in which calls this is called doesn't do anything after calling this function. Hence even if we return error here, we will just be logging it. Hence I am logging errors in this function itself.